opengrok-indexer/src/main/java/org/opengrok/indexer/history/HistoryGuru.java

```diff
@@ -491,6 +491,11 @@ public History getHistory(File file, boolean withFiles, boolean ui, boolean fall
         final File dir = file.isDirectory() ? file : file.getParentFile();
         final Repository repository = getRepository(dir);
 
+        if (repository == null) {
```

Comment:
Using [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) class instead of passing around null value is the recommended approach post java 8. 
We can go ahead with this PR and can take up a separate clean up activity later to replace this type of usage with optional.

Reply:
How would the use of `Optional` look like in this case with `getRepository()` ?

Reply:
getHistory should be returning Optional<History> instead of History.
Consumers of getHistory can invoke separate logic based upon optional value.
ex : optionalHistory.map(<logic to execute is history is not null>)
                               .orElse(<logic to execute for null case>)
https://www.youtube.com/watch?v=qrECc0COhtM is a good session about java optionals